### PR TITLE
Add protozfits recipe

### PR DESCRIPTION
**Pull request by maxnoe**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by maxnoe** [comment]
- python/c hybrid `@conda-forge/help-python-c`
- c/c++ `@conda-forge/help-c-cpp`

**Comment by maxnoe** [comment]
@conda-forge/help-c-cpp We have a CI/CD system that builds this project on Ubuntu 20.04, also using gcc 9.3. And it works.

I see that the errors are valid complaints, still, I cannot figure out why the tests are passing for us but the builds fails here.

Any idea?

**Comment by maxnoe** [comment]
@conda-forge/help-c-cpp

Ok, build is finally passing.

File: recipes/protozfits/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - zeromq
+    - protobuf
+    - zlib
+    - setuptools
+    - python
```
Comment (chrisburr) [inline recipes/protozfits/meta.yaml:33]:
```suggestion
```

File: recipes/protozfits/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: a3864c6097196208fca15b17eb49241675b3811bd6d1ffec206ce0e3bba8f89f
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [win]
+  skip: true  # [py<36]
+  missing_dso_whitelist:
+    - "$RPATH/*"
```
Comment (chrisburr) [inline recipes/protozfits/meta.yaml:23]: This shouldn't be needed:
```suggestion
```

File: recipes/protozfits/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 14 is convenient
+# when copying and pasting from another recipe, but not really needed.
```
Comment (chrisburr) [inline recipes/protozfits/meta.yaml:5]:
```suggestion
```

File: recipes/protozfits/meta.yaml (hunk 2 above)
Comment (maxnoe) [reply recipes/protozfits/meta.yaml:23]: The build fails without it. The cmake build adds the linked libraries to RPATH and some build checking script complains about not being able to find `$RPATH/libprotobuf.so`

Comment (maxnoe) [reply recipes/protozfits/meta.yaml:18]: This is needed for people installing from source with pip. If this should be deactivated in the conda forge build, what would be a nice solution?

Comment (chrisburr) [reply recipes/protozfits/meta.yaml:18]: Add `libprotobuf` to the `host` dependencies and remove it from `run`

Comment (chrisburr) [reply recipes/protozfits/meta.yaml:18]: Conda-build will then use this to automatically put an appropriate version constraint in `run`.

https://github.com/conda-forge/libprotobuf-feedstock/blob/f1471e3b75f22294942a4693d20db3d14fb3684f/recipe/meta.yaml#L47-L51

Comment (maxnoe) [reply recipes/protozfits/meta.yaml:18]: thanks!

File: recipes/protozfits/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    - python
+    - protobuf
+    - libprotobuf
+    - setuptools
+    - zeromq
+    - zlib
+  run:
+    - astropy
+    - numpy
+    - protobuf
```
Comment (chrisburr) [inline recipes/protozfits/meta.yaml:33]: I suspect this isn't needed at runtime
```suggestion
```

Comment (maxnoe) [reply recipes/protozfits/meta.yaml:33]: The `google.protobuf` package is needed at runtime.

Comment (maxnoe) [reply recipes/protozfits/meta.yaml:33]: That should be the `protobuf` package, right?

Comment (maxnoe) [reply recipes/protozfits/meta.yaml:33]: Yes, it is. So protobuf in the runtime dependencies is correct.

Comment (chrisburr) [reply recipes/protozfits/meta.yaml:33]: Indeed, sorry I didn't notice `protobuf` is the Python bindings 👍

**Comment by maxnoe** [comment]
Thanks!

**Pull request merged**